ui/store/actions.ts

```diff
@@ -1121,8 +1125,38 @@ export function removeSnap(
   snapId: string,
 ): ThunkAction<void, MetaMaskReduxState, unknown, AnyAction> {
   return async (dispatch: MetaMaskReduxDispatch) => {
-    await submitRequestToBackground('removeSnap', [snapId]);
-    await forceUpdateMetamaskState(dispatch);
+    dispatch(showLoadingIndication());
+
+    const isAccountsSnap =
+      Object.values(KEY_MANAGEMENT_SNAPS).findIndex(
+        (snap) => snap.snapId === snapId,
+      ) !== -1;
+
+    try {
+      if (isAccountsSnap) {
+        const accounts = (await handleSnapRequest({
+          snapId,
+          origin: 'metamask',
+          handler: HandlerType.OnRpcRequest,
+          request: {
+            jsonrpc: '2.0',
+            method: 'keyring_listAccounts',
```

Comment:
Do we need to ping the snap for this information? I thought they keyring kept this information?

Reply:
the keyring controller does keep this information, but it doesn't get the addresses per snap.